docs/contributing/making-a-pull-request.md

```diff
@@ -0,0 +1,450 @@
+    is a whole dark art of its own. I don't have a ton of experience with it
+    either. Perhaps something that is best figured out together - unless you
+    already have a view on this.
+
+### Testing and reviewing changes
+
+Before you commit any changes, you should make sure that they work as expected
+and do not create any unintended side effects. You should test them on at least
+these three [smoke tests]:
+
+- The documentation of Material for MkDocs itself. If you set up and run the
+development environment as outlined in the [instructions for setting up the
+development environment], `mkdocs serve` should be running and continuously
+building the documentation. Check that there are no error messages and, ideally,
+no (new) warnings.
+
+- Test on a project that represents the problem or a test for a newly developed
+feature. You may already have this if you have filed a bug report and created
+a [minimal reproduction]. If you are working on a new feature then you may need
+to build a project to serve as a test suite. It can double as documentation that
+shows how your new feature is meant to work.
+
+- Test with relevant examples from the [Material for MkDocs Examples]
+  repository. Note that to build all examples in one go you need the projects
+  plugin from Insiders but you can always build the examples individually
+  using the public version.
+
+[smoke tests]: https://en.wikipedia.org/wiki/Smoke_testing_(software)
+[minimal reproduction]: https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/
+[Material for MkDocs Examples]: https://github.com/mkdocs-material/examples
+
+- Ideally, also test the examples in the [examples repository]. If you are
+working on the Insiders edition of Material for MkDocs, you can simply start a
+build at the top level and the [projects plugin] will build all of the examples
+for you. If you are on the public version, you will need to build each
+sub-project individually. We appreciate that this is a growing collection of
+examples and you may want to prioritize those that are most relevant to the
+functionality you change.
+
+[examples repository]: https://github.com/mkdocs-material/examples
+[projects plugin]: https://squidfunk.github.io/mkdocs-material/plugins/projects/
+
+### Creating the pull request
+
+Initially, create the pull request **as a draft**. You do this [through the
+various interfaces that GitHub provides]. Which one you use is entirely up to
+you. We do not provide specific instructions for using the interfaces as GitHub
+provide all the information that should be necessary.
+
+[through the various interfaces that GitHub provides]: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request
+
+!!! question "Pull Request Template?"
+
+    @squidfunk, would it be useful to have a [PR template]? That might replace a
```

Comment:
Yes, that would be very useful. It could be similarity structured as the issue template, and link to the relevant subsections of this document to ensure they were read.

Reply:
I suggest to tackle the PR template in a separate PR.

Reply:
righto.

Reply:
Could you remove the question? Do you feel comfortable if we put this online?

Reply:
This must still be removed 😉